encodings/sequence/src/compute/compare.rs

```diff
@@ -65,6 +56,44 @@ impl CompareKernel for SequenceVTable {
     }
 }
 
+pub(crate) fn find_intersection_scalar(
+    base: PValue,
+    multiplier: PValue,
+    len: usize,
+    intercept: PValue,
+) -> Option<usize> {
+    match_each_integer_ptype!(base.ptype(), |P| {
+        let intercept = intercept
+            .as_primitive()
+            .vortex_expect("constant pvalue matching already validated");
+
+        let base = base
+            .as_primitive::<P>()
+            .vortex_expect("base pvalue matching already validated");
+        let multiplier = multiplier
+            .as_primitive::<P>()
+            .vortex_expect("multiplier pvalue matching already validated");
+
+        find_intersection(base, multiplier, len, intercept)
+    })
+}
+
+fn find_intersection<P: NativePType + PartialOrd + Eq>(
```

Comment:
Use methods from NativePType, is_eq, is_ge, is_le. It is the same definition as the builin traits but avoids people following the same patter in other places